src/Gateways/PagSeguro/CreditCard.php

```diff
@@ -0,0 +1,92 @@
+<?php
+
+namespace BeerAndCodeTeam\PaymentGateway\Gateways\PagSeguro;
+
+class CreditCard implements PaymentMethodInterface
+{
+    private int $installments;
+    private bool $capture;
+    private string $cardNumber;
+    private string $cardExpMonth;
+    private string $cardExpYear;
+    private string $cardSecurityCode;
+    private string $holderName;
+
+    public function toArray(): array
+    {
+        return [
+            'type' => 'CREDIT_CARD',
+            'installments' => $this->installments,
+            'capture' => $this->capture,
+            'card' => [
+                'number' => $this->cardNumber,
+                'exp_month' => $this->cardExpMonth,
+                'exp_year' => $this->cardExpYear,
+                'security_code' => $this->cardSecurityCode,
+                'holder' => [
+                    'name' => $this->holderName
+                ]
+            ]
+        ];
+    }
+    public function setInstallments(int $installments)
+    {
+        if ($installments < 0) {
+            throw new \Exception("Error installment less than zero ", $installments);
+        }
+        $this->installments = $installments;
+    }
+    public function setCapture(bool $capture)
+    {
+        $this->capture = $capture;
+    }
+    public function setCardNumber(string $cardNumber)
+    {
+        if (strlen($cardNumber) < 14 || 19 < strlen($cardNumber)) {
+            throw new \Exception("Error card number " . strlen($cardNumber) . " digits");
+        }
```

Comment:
Aqui a condição deveria ser um **&&** ou sempre que o tamanho da string for maior/igual a 14, a condição será verdadeira.

Reply:
Nesse caso a ideia é que o número do cartão deve ter entre 14-19 dígitos se tiver mais ou menos tem que entrar no if